crates/artifacts/zksolc/src/serde_helpers.rs

```diff
@@ -0,0 +1,61 @@
+use foundry_compilers_artifacts_solc::BytecodeObject;
+use serde::{Deserialize, Deserializer};
+
+use crate::{bytecode::Bytecode, contract::RawContract};
+
+/// Deserialize a [`RawContract`] with added `missingLibraries` field,
+/// to populate [`Bytecode`]'s `missing_libraries` with it
+pub fn maybe_unlinked_contract<'de, D>(deserializer: D) -> Result<RawContract, D::Error>
```

Comment:
I would prefer doing this when converting from contract to artifact and not when deserializing

Reply:
That sounds good